LibraBFT/Impl/Consensus/Liveness/ProposalGenerator.agda

```diff
@@ -13,6 +16,37 @@ open import Optics.All
 
 module LibraBFT.Impl.Consensus.Liveness.ProposalGenerator where
 
-postulate
-  generateProposalM : Instant → Round → LBFT (Either ErrLog BlockData)
+ensureHighestQuorumCertM : Round → LBFT (Either ErrLog QuorumCert)
+
+generateProposalM : Instant → Round → LBFT (Either ErrLog BlockData)
+generateProposalM _now round = do
+  lrg ← use (lProposalGenerator ∙ pgLastRoundGenerated)
+  ifM lrg <?ℕ round
+    then (do
+      lProposalGenerator ∙ pgLastRoundGenerated ∙= round
+      ensureHighestQuorumCertM round ∙?∙ λ hqc -> do
+        payload ← if BlockInfo.hasReconfiguration (hqc ^∙ qcCertifiedBlock)
```

Comment:
I will go ahead an add `ifM`, `caseMM` where appropriate.

Reply:
Super (and I will keep them in mind in future work).